src/Deserializers/EntityDeserializer.php

```diff
@@ -99,8 +100,8 @@ private function setDescriptionsFromSerialization( array $serialization, Entity
 	private function deserializeValuePerLanguageSerialization( array $serialization ) {
 		$array = array();
 
-		foreach( $serialization as $valueSerialization ) {
-			$this->assertIsValidValueSerialization( $valueSerialization );
+		foreach( $serialization as $requestedLanguage => $valueSerialization ) {
```

Comment:
For the record: This change makes the deserialization fail if the serialization does not have array keys. This is by design, as far as I understand.